.github/rulesets/apply-rulesets.sh

```
@@ -0,0 +1,190 @@
+check_auth() {
+    if ! gh auth status &> /dev/null; then
+        echo -e "${RED}❌ Not authenticated with GitHub CLI.${NC}"
+        echo "Please run: gh auth login"
+        exit 1
+    fi
+}
+
+# Function to apply a ruleset
+apply_ruleset() {
+    local ruleset_file=$1
+    local ruleset_name=$(basename "$ruleset_file" .json)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)